src/Symfony/Component/HttpFoundation/Session/Flash/FlashBagInterface.php

```diff
@@ -24,7 +24,7 @@
      * Adds a flash message for type.
      *
      * @param string $type
-     * @param string $message
+     * @param mixed  $message
```

Comment:
IIRC the rationale behind this type hint is that Symfony does not (did not?) want to take the responsability of serializing this properly. So technically, it works, but that might generate invalid support.

Reply:
If this still holds true, maybe we should add a comment right here, explaining just that.

Reply:
not sure how this applies: you are already allowed to put any serializable in the session, do we have special comments about this?

Reply:
I can't find the source of this discussion I saw, but if we go this way, I'd suggest adding a message indicating that `$message` should be serializable, and what that means (it doesn't crash when given to `\serialize()`? is the symfony serializer supported?)

Reply:
Also, a docs PR should probably be done too.

Reply:
`FlashBagInterface::add` should have exactly the same support as `Session::set` (it's basicaly a specialized `Session::set`).

`Session::set` doesn't say anything about what can be stored in the session (https://github.com/symfony/symfony/blob/master/src/Symfony/Component/HttpFoundation/Session/SessionInterface.php#L121) so I gess it's the responsibility of the implementation to decide what can be stored (all the implementations in symfony are based on `serialize` but some libs could extend it to support something else).

Reply:
Forget everything I said then, it would make no sense to add that comment here and not in the session interface.